packages/fuels-programs/src/contract.rs

```diff
@@ -603,21 +603,8 @@ where
         Ok(SubmitResponse::new(tx_id, self))
     }
 
-    pub async fn response(self) -> Result<FuelCallResponse<D>> {
```

Comment:
From what I understand, this was meant as a shortcut when using `submit()` instead of blocking on `call()`. The alternative would be to use `respnse()` on the `SubmitResponse` returned from `submit()` which is guaranteed to have a tx_id internally.

I don't think that's a bad idea, fetching a response from a response feels kind of awkward. Maybe we can discuss how to make the UX around this friendlier.
`get_response` suffers from a similar problem in that it will just use `None` as the tx_id in the response if no call was made.

Reply:
We can not call `response` after `submit` as `submit` takes ownership and we have no way to access the `cached_tx_id` directly. 

I agree that we need to improve the UX and refactor this part of the code. I initially wanted to do it in this PR but I saw that it would take more time and effort and decided to only remove the redundant method.